software/base/src/main/java/org/apache/brooklyn/entity/software/base/lifecycle/MachineLifecycleEffectorTasks.java

```diff
@@ -709,9 +710,11 @@ protected void doStop(ConfigBag parameters, Callable<StopMachineDetails<Integer>
         DynamicTasks.queue("pre-stop", new PreStopCustomTask());
 
         Maybe<MachineLocation> machine = Machines.findUniqueMachineLocation(entity().getLocations());
-        Task<String> stoppingProcess = null;
+        Task<List<?>> stoppingProcess = null;
         if (canStop(stopProcessMode, entity())) {
-            stoppingProcess = DynamicTasks.queue("stopping (process)", new StopProcessesAtMachineTask());
+            stoppingProcess = Tasks.parallel(
```

Comment:
Aled and I debugged the hanging test today and narrowed it down to this line. The stoppingProcess task is created, but never started.

The following code can be used instead:
`
stoppingProcess = Tasks.parallel(
                    Tasks.<String>builder().displayName("stopping (process)").body(new StopProcessesAtMachineTask()).build(),
                    Tasks.<String>builder().displayName("stopping (feeds)").body(new StopFeedsAtMachineTask()).build());
            DynamicTasks.queue(stoppingProcess);
`